### ISSUE TITLE
[New Rule] Default values for dictionaries

### ISSUE THREAD

**Issue by MartinThoma** [MartinThoma/flake8-simplify#80]
## Explanation

It's a pretty typical situation that you have a dictionary and you need to do something with a value in there. If the key is not in there, you fill it (with another dict / a list / something else).

This rule simplifies the code as it removes code duplication.

## Example

```python
# Bad
if key in my_dict:
    my_dict = my_dict[key]
else:
    my_dict[key] = {}
    my_dict = my_dict[key]

# Good
if key not in my_dict:
    my_dict[key] = {}
my_dict = my_dict[key]
```

**Comment by TomFryers** [comment]
Or, even better, `my_dict = my_dict.setdefault(key, {})`.
